File: files/CH-126.R
library(tidyverse)
library(readxl)

path = "files/CH-126 Transformation.xlsx"
input = read_excel(path, range = "B2:C8")
test  = read_excel(path, range = "E2:G20") %>%
  mutate(Dates = as.Date(Dates))

result = input %>%
  separate_wider_delim(Dates, delim = ", ", names = c("Registeration", "Evaluation", "Approved")) %>%
  pivot_longer(cols = -`Order IDS`, names_to = "Group", values_to = "Dates") %>%
  mutate(Dates = as.Date(Dates)) %>%
  arrange(`Order IDS`, Dates)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: files/100-199/126/CH-126.py
import pandas as pd

path = "CH-126 Transformation.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="E:G", skiprows=1, nrows=19).rename(columns=lambda x: x.replace('.1', ''))
test['Dates'] = pd.to_datetime(test['Dates'])

result = input\
          .assign(Dates=input['Dates'].str.split(', '))\
          .explode('Dates')\
          .assign(Dates=lambda df: pd.to_datetime(df['Dates']))\
          .assign(rownumber=lambda df: df.groupby('Order IDS').cumcount() + 1)\
          .assign(Group=lambda df: df['rownumber'].map({1: 'Registeration', 2: 'Evaluation', 3: 'Approved'}))\
          .drop(columns='rownumber')\
          .sort_values(by = ['Order IDS', 'Dates'])\
          .reset_index(drop=True)
result = result[['Order IDS', 'Group', 'Dates']]

print(result.equals(test)) # True